### ISSUE TITLE
Code for attentive pooling for CNN

### ISSUE THREAD

**Issue by akjindal53244** [lancopku/AAPR#5]
Dear Authors,

Thanks for your work. It was really interesting to learn something new :)

I have a doubt regarding attentive pooling for CNN. There are two classes `class Attention` and `class Conv_attention`. I couldn't find any uses of `class Conv_attention` however.

I guess the implementation for attentive pooling is inside `def forward(self, x)` defined in `class Attention`

**code of `def forward()`:**
`u = self.linear_project(x.contiguous().view(-1, input_size)).contiguous().view(batch_size, seq_len, -1)
`
`atten_weights = self.softmax(batch_matmul(u, self.representation))
`
`s = attention_mul(x, atten_weights)
`
`return s
`

<img width="232" alt="screen shot 2018-08-22 at 12 06 32 am" src="https://user-images.githubusercontent.com/5215386/44421850-4e7d0780-a59f-11e8-9e44-358bb7cfa435.png">

In equation `tanh` is being applied after linear transformation followed by attention calculation which involves multiplication with another vector (u_w).
while in method `batch_matmul()`, it seems that `tanh` is being called after multiplication with `u_w`. 

Is it right? I am not much familiar with torch.
If so, how both would yield the same results?

Can you please provide equivalent `tensorflow` code for the same?

**Comment by hanxiao** [comment]
Here is [a Tensorflow implementation](https://github.com/hanxiao/tf-attentive-conv) of Yin Wenpeng's recent paper on TACL "Attentive Convolution". The author argue that it is much more effective than attentive pooling on CNN.

Btw, it is also part of a bigger project: [tf-nlp-blocks](https://github.com/hanxiao/tf-nlp-blocks)
